src/components/configuration/CockpitSettingsManager.vue

```diff
@@ -0,0 +1,400 @@
+}
+
+const finishInlineJsonEditing = (key: string): void => {
+  try {
+    const parsed = JSON.parse(inlineJsonText[key])
+    editedValues[key] = parsed
+    inlineJsonEditing[key] = false
+    JsonEditError.value = false
+    userSettings.value[key] = parsed
+  } catch (error: any) {
+    JsonEditError.value = true
+    openSnackbar({ message: 'Invalid JSON: ' + error.message, variant: 'error', duration: 5000 })
+  }
+}
+
+const cancelJsonEditing = (): void => {
+  Object.keys(inlineJsonEditing).forEach((key) => {
+    inlineJsonEditing[key] = false
+  })
+  JsonEditError.value = false
+}
+
+const downloadConfigFile = (): void => {
+  const dataStr = JSON.stringify(userSettings.value, null, 2)
+  const blob = new Blob([dataStr], { type: 'application/json' })
+  const url = URL.createObjectURL(blob)
+  const a = document.createElement('a')
+  const username = missionStore.username !== 'null' ? missionStore.username : 'unnamed'
+  const filename = `${username}_config.json`
+  a.href = url
+  a.download = filename
+  document.body.appendChild(a)
+  a.click()
+  document.body.removeChild(a)
+  URL.revokeObjectURL(url)
+}
+
+const uploadConfigFile = (): void => {
+  const input = document.createElement('input')
+  input.type = 'file'
+  input.accept = 'application/json'
+  input.onchange = (event: Event) => {
+    const file = (event.target as HTMLInputElement).files?.[0]
+    if (!file) return
+    const reader = new FileReader()
+    reader.onload = (e: ProgressEvent<FileReader>) => {
+      try {
+        const text = e.target?.result as string
+        const json = JSON.parse(text)
+        if (!json || typeof json !== 'object') {
+          openSnackbar({
+            message: 'Invalid configuration file: content is not an object.',
+            variant: 'warning',
+            duration: 5000,
+          })
+          return
+        }
+        Object.assign(editedValues, json)
+        userSettings.value = json
+        openSnackbar({ message: 'Configuration file applied successfully.', variant: 'success', duration: 5000 })
+      } catch (error: any) {
+        console.error('Error parsing configuration file: ' + error.message)
+        openSnackbar({
+          message: 'Error parsing configuration file: ' + error.message,
+          variant: 'error',
+          duration: 5000,
+        })
+      }
+    }
+    reader.readAsText(file)
+  }
+  input.click()
+}
+
+const resetAllCockpitSettings = (): void => {
+  showDialog({
+    message:
+      "Are you sure you want to reset Cockpit's browser stored settings to defaults? " +
+      'Settings on the vehicle will not be affected. ' +
+      'Once you connect back to the vehicle, the settings will be restored.',
+    variant: 'warning',
+    maxWidth: 800,
+    actions: [
+      {
+        text: 'Cancel',
+        action: () => closeDialog(),
+      },
+      {
+        text: 'Reset settings',
+        action: () => {
+          localStorage.clear()
+          location.reload()
```

Comment:
This will reload the page immediately, so the snackbar won't show or will do for a quick moment.

There's a utils method that deals with warning about the reload and setting a timeout so the user can read any message left. It can help here.